### DISCUSSION TITLE
全局搜索功能不好使

### DISCUSSION THREAD

**Discussion by aaja** [TriDiamond/hexo-theme-obsidian#123] [Q&A] [answered]
按照文档的说明
npm install hexo-generator-searchdb --save
然后主项目的_config.yml也配置了，不行，请求大佬帮助！！！万分感谢

**Comment by bennyxguo** [comment]
你要看看public文件夹下有没有生成`search.xml`文件。
![image](https://user-images.githubusercontent.com/24708758/87669053-dc6bc800-c79f-11ea-8dd5-8f366692758d.png)

**Comment by aaja** [comment]
有生成search.xml但是文件比较大1M多，是不是因为文件太大的问题呢

**Comment by kasie-zhang** [comment, marked as answer]
提供一种解决思路：

## 查看问题

 打开浏览器，查看search功能是否引发了什么错误

控制台中可以看到search功能引发的错误是无法找到 search.xml 文件

![](https://sherrykeeper.oss-cn-hangzhou.aliyuncs.com/md/md_056.png)

但是 build 完成后，项目中已经生成 search.xml 文件，所以一定是某个 js 文件出错。



## 解决问题

在 themes\obsidian\js 目录下找到了 search.js 文件，该文件控制实现 search功能。

查看该文件下与 path 相关的变量，可以找到 getSearchFile 函数

```js
var getSearchFile = function(){
    var path = "/search.xml";
    searchFunc(path, 'local-search-input', 'local-search-result');
};
```

将 path 变量更改为适合的路径：

![](https://sherrykeeper.oss-cn-hangzhou.aliyuncs.com/md/md_057.png)

## 验证是否成功

进入博客页面验证 search 功能是否可用！

![](https://sherrykeeper.oss-cn-hangzhou.aliyuncs.com/md/md_058.png)



完结，撒花 :rainbow: :cherry_blossom:

**Comment by bennyxguo** [comment]
@hei-lou 可以打开控制台，看看有没有报错吗？
